jobs/periodic/notification-courses-ended-yesterday/index.ts

```diff
@@ -0,0 +1,57 @@
+import { prisma } from '../../../common/prisma';
+import moment from 'moment';
+import { getLogger } from '../../../common/logger/logger';
+import { getNotificationContextForSubcourse } from '../../../common/mails/courses';
+import { userForPupil, userForStudent } from '../../../common/user';
+import * as Notification from '../../../common/notification';
+
+const logger = getLogger();
+
+export default async function execute() {
+    const subcoursesWithLecturesEndedYesterday = await prisma.subcourse.findMany({
+        where: {
+            lecture: {
+                some: {
+                    start: {
+                        gte: moment().subtract(1, 'day').startOf('day').toDate(),
+                        lt: moment().startOf('day').toDate(),
+                    },
+                },
+            },
+        },
+        include: {
+            course: true,
+            lecture: true,
+            subcourse_instructors_student: { include: { student: true } },
+            subcourse_participants_pupil: { include: { pupil: true } },
+        },
+    });
+    for (const subcourse of subcoursesWithLecturesEndedYesterday) {
+        if (subcourse.lecture.length === 0) {
+            continue;
+        }
+        const lastLecture = subcourse.lecture.sort((a, b) => a.start.getTime() - b.start.getTime())[subcourse.lecture.length - 1];
+        if (
+            moment(lastLecture.start).add(lastLecture.duration, 'seconds').toDate() >= moment().subtract(1, 'day').startOf('day').toDate() &&
```

Comment:
Why add duration?

Reply:
If a lecture (hypothetically) crosses midnight, we would skip it here

Reply:
I don't think they need to be "correct" in the corner case that a lecture happens at midnight, but the predicate here should match the predicate in the query.